### PR TITLE
Issue #461: Extract version wherever possible with python3 setup.py --version

### LINKED ISSUE #461: Extract version wherever possible with `python3 setup.py --version` rather then `grep ... | cut ...`
See e.g. `ci/deb-ppa-upload/build-context/entrypoint.sh` -> `VERSION=$(cut -d\' -f2 git_machete/__init__.py)`

### PR DESCRIPTION
**Pull request by amalota**
closes #461

### CODE REVIEW

File: ci/nixpkgs-build/build-context/entrypoint.sh [hunk 1]
```diff
@@ -5,7 +5,7 @@ set -e -o pipefail -u -x
 git clone --depth=1 https://github.com/NixOS/nixpkgs.git .
 
 source_hash=$(nix-prefetch-url --unpack https://github.com/VirtusLab/git-machete/archive/$GIT_REVISION.tar.gz)
-version=$(curl https://raw.githubusercontent.com/VirtusLab/git-machete/$GIT_REVISION/git_machete/__init__.py | cut -d\' -f2)
+version=$(curl https://raw.githubusercontent.com/VirtusLab/git-machete/$GIT_REVISION/git_machete/__init__.py | cut -d\' -f2) #NOTE: NOT SURE HOW THIS CAN BE DONE
```
Comment (amalota) [inline ci/nixpkgs-build/build-context/entrypoint.sh]: @PawelLipski Need some help with this one.

File: hook_samples/machete-status-branch [hunk 2]
```diff
@@ -16,4 +16,4 @@ function ansi_esc {
 dim=$(ansi_esc '\033[2m')
 endc=$(ansi_esc '\033[0m')
 
-echo -ne "${dim}v$(git grep "__version__ = " "$1" -- git_machete/__init__.py | cut -d\' -f2 || echo '???')${endc}"
+echo -ne "${dim}v$(git grep "__version__ = " "$1" -- git_machete/__init__.py | cut -d\' -f2 || echo '???')${endc}" #NOTE: NOT SURE WHATS HAPPENING HERE AND HOW THIS CAN BE CHANGED
```
Comment (amalota) [inline hook_samples/machete-status-branch]: @PawelLipski Need some help with this one too.

File: ci/nixpkgs-build/build-context/entrypoint.sh (hunk 1 above)
Comment (PawelLipski) [reply ci/nixpkgs-build/build-context/entrypoint.sh]: goncern

Comment (PawelLipski) [reply ci/nixpkgs-build/build-context/entrypoint.sh]: A fix worse than the problem itself would be to check out the repo at `$GIT_REVISION` and then run `python3 setup.py --version` on that repo...

Comment (PawelLipski) [reply ci/nixpkgs-build/build-context/entrypoint.sh]: Let's leave as is! the current approach is simpler than the alternative

File: hook_samples/machete-status-branch (hunk 2 above)
Comment (PawelLipski) [reply hook_samples/machete-status-branch]: Whoops that's a similar problem as with `ci/nixpkgs-build/build-context/entrypoint.sh`... a version is extracted here NOT from the current state of the repo, but instead from the given branch (hence `$1` is passed as the commit argument to `git grep`: `git grep <pattern> <commit> -- <files>`).

Comment (PawelLipski) [reply hook_samples/machete-status-branch]: So... let's leave as is here as well

**Review by PawelLipski — APPROVED** [review]

**Pull request merged**